Policies/Kubernetes/container-no-privilege-escalation/rego/src_test.rego

```diff
@@ -0,0 +1,140 @@
+package k8sazurecontainernoprivilegeescalation
+
+test_input_container_not_privilege_escalation_allowed {
+    input := { "review": input_review}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_container_privilege_escalation_not_allowed {
+    input := { "review": input_review_priv}
+    results := violation with input as input
+    count(results) == 1
+}
+test_input_container_many_not_privilege_escalation_allowed {
+    input := { "review": input_review_many}
+    results := violation with input as input
+    count(results) == 2
```

Comment:
As per my understanding of thi spolicy, violation should be listed if allowedPrivilegeEscalation property is set to true in securityContext. I see input_review_many includes input_containers_many and input_containers_one. I see containers in input_containers_many and input_containers_one do not have any violation. Why is the count(results) value is 2 here?

Reply:
This policy raises a violation if the allowPrivilegeEscalation property is not explicitly set to false, input_containers_many will raise a violation with the second and third containers

Reply:
@sozercan Should we show violation if allowPrivilegeEscalation property not set on a container? I assume this property is not used in general by most of the conatiners which means we flag all of them.

Reply:
if `allowPrivilegeEscalation` is not set, k8s will default to true to not break existing binaries (https://kubernetes.io/docs/concepts/policy/pod-security-policy/#privilege-escalation)
PSP can set `DefaultAllowPrivilegeEscalation` to mutate resources to change defaults (which GK doesn't support since it doesn't support mutation yet).

so if our constraint is to ensure `allowPrivilegeEscalation` is set to `false` then we should show violation if its not set